README.md

```diff
@@ -1,95 +1,150 @@
 ```bash
 cd LLM
 pip install -r requirements.txt
 ```
 
+If you have any errors, please create a new conda environment and install Pytorch following this page: https://pytorch.org/get-started/locally/
+
 ### Running
 
 ```bash
-conda activate imposter
 python run.py
 ```
+Remember: you MUST be in the `imposter` conda context, and in the LLM directory, as a large number of packages required by the LLM are installed 'contextually' in the conda environment we created earlier. You can switch the conda context anytime with `conda activate imposter`. If your terminal does not show something like `(imposter)` at the start, and instead shows nothing or something like `(base)`, you are not in the right conda environment.
 
-## TTS server
+### Optional: LLM customisations
+At the point of `python run.py`, you can add additional params to suit your needs. For example:
 
-### GPU version
+Use `--load-8bit` if you get CUDA memory errors (out of memory), this will try to load an 8bit variant of the model. Will be less impressive.
 ```bash
-docker run --rm -it -p 5002:5002 --gpus all -v /c/vosk:/root/.local/share/ --entrypoint /bin/bash ghcr.io/coqui-ai/tts:v0.11.1
-python3 TTS/server/server.py --model_name tts_models/en/vctk/vits --use_cuda true
+python run.py --load-8bit
 ```
 
-### CPU version
-``` 
-docker run --rm -it -p 5002:5002 --entrypoint /bin/bash ghcr.io/coqui-ai/tts-cpu:v0.11.1
-python3 TTS/server/server.py --model_name tts_models/en/vctk/vits
+Use --num-gpus (some number) if you want to maintain the quality and have multiple GPUs available:
+```bash
+python run.py --num-gpus 2
 ```
 
+Use --device cpu if you have enough RAM but your GPU can't support even the --load-8bit, or you want to maintain quality. This will be extremely slow.
+```bash
+python run.py --device cpu
+```
 
-Github repository: https://github.com/coqui-ai/TTS
+Use --model-name some/huggingfacemodel to change the model! Note, you may have to edit the `tokenizer = blahblah` line in server.js based on what tokenizer the huggingface model page recommends (especially if errors or strange behaviour)
+```bash
+python run.py --model-name EleutherAI/gpt-neox-20b
+```
 
-## STT server
+Use --max-new-tokens (with a default of 256) to increase how many tokens are available (and thus how long a conversation can go for)
 ```bash
-docker run --rm --name=sepia-stt -p 20741:20741 -it sepia/stt-server:dynamic_v1.0.0_amd64
+python run.py --max-new-tokens 1024
 ```
-Github repository: https://github.com/SEPIA-Framework/sepia-stt-server
 
-## Speech server
+Use --temperature (with a default of 0.7) to increase or decrease the randomness/stability
 ```bash
-node speech.js
+python run.py --temperature 0.5
 ```
 
-# Credits
+Most of the above params can be added together, eg python run.py --temperature 0.5 --max-new-tokens 1024 --model-name EleutherAI/gpt-neox-20b
 
-- @Acidbubbles for their perfect code template: https://github.com/acidbubbles/vam-plugin-template
-- @VAMDeluxe for their amazing code.
```

Comment:
We do not want to remove credits, can you check if you have successfully rebased on `main`?

Reply:
Oh, I see that had some trouble with git, I think the only missing thing is my last commit on main where I added the credits section.

Reply:
Oh dear I genuinely did NOT want to delete them! At this point I was working in a terminal window and must have accidentally removed that. Good spot, thank you.